genesyscloud/provider/provider.go

```diff
@@ -268,14 +268,23 @@ func InitClientConfig(data *schema.ResourceData, version string, config *platfor
 		RetryWaitMax: time.Second * 30,
 		RetryMax:     20,
 		RequestLogHook: func(request *http.Request, count int) {
-			if count > 0 && request != nil {
-				log.Printf("Retry #%d for %s %s", count, request.Method, request.URL)
+			sdkDebugRequest := newSDKDebugRequest(request, count)
+			request.Header.Set("TF-Correlation-Id", sdkDebugRequest.TransactionId)
```

Comment:
I am setting this variable on the request so we can capture it on the response (even though the public API does not give it back).  By having a stable ID, I can then merge the request and response in my analysis on a single call.